core/bessctl.cc

```diff
@@ -550,17 +573,26 @@ class BESSControlImpl final : public BESSControl::Service {
       });
     }
 
+    if (request->wid() < 0) {
```

Comment:
This block is meant to catch the case where no specific wid was requested (request->wid()==-1).

request->wid() is saved into wid_filter, but then wid_filter is altered:

```
    if (wid_filter < 0) {
      i = 0;
      wid_filter = num_workers - 1;
    } else {
      i = wid_filter;
    }
```